### ISSUE TITLE
ClassCastException on upgrading Enunciate 1.30 project to 2.0.0-M1

### ISSUE THREAD

**Issue by marstonstudio** [stoicflame/enunciate#97]
When trying to run Enunciate using Maven on my Jackson + Jersey project I get the following exception. Project works with Enunciate 1.30

```
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 31.982s
[INFO] Finished at: Wed Jul 22 16:27:40 EDT 2015
[INFO] Final Memory: 88M/627M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal com.webcohesion.enunciate:enunciate-maven-plugin:2.0.0-M.1:assemble (default) on project reportcard: Execution default of goal com.webcohesion.enunciate:enunciate-maven-plugin:2.0.0-M.1:assemble failed: java.lang.ClassCastException: com.sun.tools.javac.code.Type$ClassType cannot be cast to com.webcohesion.enunciate.javac.decorations.type.DecoratedTypeMirror: OnError while emitting onNext value: com.webcohesion.enunciate.EnunciateContext.class -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal com.webcohesion.enunciate:enunciate-maven-plugin:2.0.0-M.1:assemble (default) on project reportcard: Execution default of goal com.webcohesion.enunciate:enunciate-maven-plugin:2.0.0-M.1:assemble failed: java.lang.ClassCastException: com.sun.tools.javac.code.Type$ClassType cannot be cast to com.webcohesion.enunciate.javac.decorations.type.DecoratedTypeMirror
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:224)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:153)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:145)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:84)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:59)
	at org.apache.maven.lifecycle.internal.LifecycleStarter.singleThreadedBuild(LifecycleStarter.java:183)
	at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:161)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:317)
	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:152)
	at org.apache.maven.cli.MavenCli.execute(MavenCli.java:555)
	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:214)
	at org.apache.maven.cli.MavenCli.main(MavenCli.java:158)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:606)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:289)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)
	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)
	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)
Caused by: org.apache.maven.plugin.PluginExecutionException: Execution default of goal com.webcohesion.enunciate:enunciate-maven-plugin:2.0.0-M.1:assemble failed: java.lang.ClassCastException: com.sun.tools.javac.code.Type$ClassType cannot be cast to com.webcohesion.enunciate.javac.decorations.type.DecoratedTypeMirror
	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:115)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:208)
	... 19 more
Caused by: java.lang.RuntimeException: java.lang.ClassCastException: com.sun.tools.javac.code.Type$ClassType cannot be cast to com.webcohesion.enunciate.javac.decorations.type.DecoratedTypeMirror
	at com.sun.tools.javac.main.Main.compile(Main.java:469)
	at com.sun.tools.javac.api.JavacTaskImpl.call(JavacTaskImpl.java:132)
	at com.webcohesion.enunciate.Enunciate.run(Enunciate.java:585)
	at com.webcohesion.enunciate.mojo.ConfigMojo.execute(ConfigMojo.java:304)
	at com.webcohesion.enunciate.mojo.AssembleBaseMojo.execute(AssembleBaseMojo.java:41)
	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:106)
	... 20 more
Caused by: java.lang.ClassCastException: com.sun.tools.javac.code.Type$ClassType cannot be cast to com.webcohesion.enunciate.javac.decorations.type.DecoratedTypeMirror
	at com.webcohesion.enunciate.modules.jackson.EnunciateJacksonContext.isKnownType(EnunciateJacksonContext.java:363)
	at com.webcohesion.enunciate.modules.jackson.EnunciateJacksonContext.isKnownTypeDefinition(EnunciateJacksonContext.java:272)
	at com.webcohesion.enunciate.modules.jackson.EnunciateJacksonModule.addPotentialJacksonElement(EnunciateJacksonModule.java:95)
	at com.webcohesion.enunciate.modules.jackson.EnunciateJacksonModule.addDataTypeDefinition(EnunciateJacksonModule.java:86)
	at com.webcohesion.enunciate.modules.jaxrs.EnunciateJaxrsModule.addReferencedDataTypeDefinitions(EnunciateJaxrsModule.java:124)
	at com.webcohesion.enunciate.modules.jaxrs.EnunciateJaxrsModule.call(EnunciateJaxrsModule.java:80)
	at com.webcohesion.enunciate.io.InvokeEnunciateModule.onNext(InvokeEnunciateModule.java:31)
	at com.webcohesion.enunciate.io.InvokeEnunciateModule.onNext(InvokeEnunciateModule.java:10)
	at rx.internal.operators.OperatorDoOnEach$1.onNext(OperatorDoOnEach.java:79)
	at rx.internal.operators.OperatorSingle$ParentSubscriber.onCompleted(OperatorSingle.java:109)
	at rx.internal.operators.NotificationLite.accept(NotificationLite.java:140)
	at rx.internal.operators.TakeLastQueueProducer.emit(TakeLastQueueProducer.java:100)
	at rx.internal.operators.TakeLastQueueProducer.startEmitting(TakeLastQueueProducer.java:45)
	at rx.internal.operators.OperatorTakeLast$1.onCompleted(OperatorTakeLast.java:59)
	at rx.internal.operators.OperatorMerge$MergeSubscriber.drainAndComplete(OperatorMerge.java:521)
	at rx.internal.operators.OperatorMerge$MergeSubscriber.completeInner(OperatorMerge.java:486)
	at rx.internal.operators.OperatorMerge$InnerSubscriber.emit(OperatorMerge.java:658)
	at rx.internal.operators.OperatorMerge$InnerSubscriber.onCompleted(OperatorMerge.java:600)
	at rx.internal.operators.OperatorMerge$MergeSubscriber.drainAndComplete(OperatorMerge.java:521)
	at rx.internal.operators.OperatorMerge$MergeSubscriber.completeInner(OperatorMerge.java:486)
	at rx.internal.operators.OperatorMerge$InnerSubscriber.emit(OperatorMerge.java:673)
	at rx.internal.operators.OperatorMerge$InnerSubscriber.onCompleted(OperatorMerge.java:600)
	at rx.internal.util.ScalarSynchronousObservable$ScalarSynchronousAction.call(ScalarSynchronousObservable.java:117)
	at rx.schedulers.ExecutorScheduler$ExecutorAction.run(ExecutorScheduler.java:173)
	at rx.schedulers.ExecutorScheduler$ExecutorSchedulerWorker.run(ExecutorScheduler.java:99)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at java.lang.Thread.run(Thread.java:745)
Caused by: rx.exceptions.OnErrorThrowable$OnNextValue: OnError while emitting onNext value: com.webcohesion.enunciate.EnunciateContext.class
	at rx.exceptions.OnErrorThrowable.addValueAsLastCause(OnErrorThrowable.java:101)
	at rx.internal.operators.OperatorDoOnEach$1.onNext(OperatorDoOnEach.java:81)
	... 19 more
[ERROR] 

```

**Comment by stoicflame** [comment]
Thanks a ton for the report. I fixed this at c95b115. I don't currently have a workaround. I'll release a M.2 early next week for you to try again.

**Comment by marstonstudio** [comment]
Happy to help. 

We rely on enunciate for our API at work, it's a nice balance of a documentation tool that helps with what we need without taking over the project. Thanks for putting in the time to upgrade to Java 8.
